projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/deploy/SupportedPythonVersions.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2023-2023 VMware, Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.vmware.taurus.service.deploy;
+
+import java.security.InvalidParameterException;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import org.springframework.beans.factory.annotation.Value;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Handles operations related to supported python versions for data job deployments. The class
+ * contains utility methods which provide functionality to read the configuration related to the
+ * python versions supported by the Control Service. These utility methods are meant to be used in
+ * other components, as needed.
+ */
+@Component
+@RequiredArgsConstructor
+@Slf4j
+public class SupportedPythonVersions {
+
+  @Value("#{${datajobs.deployment.supportedPythonVersions}}")
+  private Map<String, Map<String, String>> supportedPythonVersions;
```

Comment:
I think it would be a bit nicer if you made a record for the second map. 

```
public record DeploymentDetails (String vdkImage, String baseImage) {}

private Map<String, DeploymentDetails> supportedPythonVersions;
```


It would let you remove lines 34-36.
on lines 104,114, 80 and 90 you wouldn't need to worry about the keys being missing. 
Plus I also think it documents the class a bit better. 

But I actually don't know would spring deserialise that from config so natively.

Reply:
If this is too much work you can ignore and just mark the comment as resolved.

Reply:
Done, thanks